scripts/o3de/tests/test_export_project.py

```diff
@@ -0,0 +1,157 @@
+#
+# Copyright (c) Contributors to the Open 3D Engine Project.
+# For complete copyright and license terms please see the LICENSE at the root of this distribution.
+#
+# SPDX-License-Identifier: Apache-2.0 OR MIT
+#
+#
+
+import argparse
+import pytest
+import pathlib
+import unittest.mock as mock
+from unittest.mock import patch
+from o3de import utils, manifest
+from o3de.export_project import O3DEScriptExportContext, execute_python_script, add_args, _export_script, process_command
+
+TEST_PROJECT_JSON_PAYLOAD = '''
+{
+    "project_name": "TestProject",
+    "project_id": "{24114e69-306d-4de6-b3b4-4cb1a3eca58e}",
+    "version" : "0.0.0",
+    "compatible_engines" : [
+        "o3de-sdk==2205.01"
+    ],
+    "engine_api_dependencies" : [
+        "framework==1.2.3"
+    ],
+    "origin": "The primary repo for TestProject goes here: i.e. http://www.mydomain.com",
+    "license": "What license TestProject uses goes here: i.e. https://opensource.org/licenses/MIT",
+    "display_name": "TestProject",
+    "summary": "A short description of TestProject.",
+    "canonical_tags": [
+        "Project"
+    ],
+    "user_tags": [
+        "TestProject"
+    ],
+    "icon_path": "preview.png",
+    "engine": "o3de-install",
+    "restricted_name": "projects",
+    "external_subdirectories": [
+        "D:/TestGem"
+    ]
+}
+'''
+
+#Note: the underlying command logic is found in CLICommand class object. That is tested in test_utils.py
+@pytest.mark.parametrize("args, expected_result",[
+    pytest.param(["cmake", "--version"], 0),
+    pytest.param(["cmake"], 0),
+    pytest.param(["cmake", "-B"], 1),
+    pytest.param([], 1),
+])
+def test_process_command(args, expected_result):
+
+    cli_command = mock.Mock()
+    cli_command.run.return_value = expected_result
+
+    with patch("o3de.utils.CLICommand", return_value=cli_command) as cli:
+        result = process_command(args)
+        assert result == expected_result
+
+
+
+#The following functions will do integration tests of _export_script and execute_python_script, thereby testing all of script execution
+TEST_PYTHON_SCRIPT = """
+import pathlib
+
+folder = pathlib.Path(__file__).parent
+
+with open(folder / "test_output.txt", 'w') as test_file:
+    test_file.write(f"This is a test for the following: {o3de_context.args[0]}")
+
+    """
+
+@pytest.mark.parametrize("args, should_pass_project_folder, project_folder_subpath, script_folder_subpath, output_filename, expected_result", [
+    #successful cases
+    pytest.param(['456'], False, ["test_project"], ["test_project" , "ExportScripts"], "test_output.txt", 0),
```

Comment:
I think it would be clear to make the script_folder_subpath an actual pathlib.PurePath.
```suggestion
@pytest.mark.parametrize("args, should_pass_project_folder, project_folder_subpath, script_folder_subpath, output_filename, expected_result", [
    #successful cases
    pytest.param(['456'], False, pathlib.PurePath("test_project"), pathlib.PurePath("test_project/ExportScripts"), pathlib.PurePath("test_output.txt"), 0),
```